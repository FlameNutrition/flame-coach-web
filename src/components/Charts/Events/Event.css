.complex-swipeable-list-item {
    background-color: white;
    padding: 10px;
    flex: 1;
    min-width: 0;
    user-select: none;
    cursor: pointer;
}

.complex-swipeable-list-item-label {
    display: flex;
    align-items: center;
}

.complex-swipeable-list-item-name {
    font-weight: bold;
    font-family: "Roboto", "Helvetica", "Arial", sans-serif;
    font-size: 19px;
    line-height: 27px;
    color: #162473;
    margin-left: 4px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}

.complex-swipeable-list-item-description {
    margin-left: 2px;
    margin-top: 4px;
    color: #000000;
    font-family: "Roboto", "Helvetica", "Arial", sans-serif;
    font-size: 14px;
    line-height: 17px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}
