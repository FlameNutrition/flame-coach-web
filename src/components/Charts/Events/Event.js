import React from 'react';
import './Event.css';
import PropTypes from 'prop-types';
import { Paper } from '@material-ui/core';
import moment from 'moment';

const Event = ({
  date, value, measureType
}) => {
  return (
    <Paper
      className="complex-swipeable-list-item"
      variant="outlined"
      square
    >
      <div className="complex-swipeable-list-item-label">
        <span className="complex-swipeable-list-item-name">
          {value}
          {' '}
          {measureType}
        </span>
      </div>

      <div className="complex-swipeable-list-item-description">
        {moment(date).format('ddd, MMMM Do YYYY')}
      </div>

    </Paper>
  );
};

Event.propTypes = {
  date: PropTypes.string.isRequired,
  value: PropTypes.number.isRequired,
  measureType: PropTypes.string
};

Event.defaultProps = {
  measureType: 'kg'
};

export default Event;
