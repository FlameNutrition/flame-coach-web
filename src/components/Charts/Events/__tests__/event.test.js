import React from 'react';
import { render, screen } from '../../../../testing/test-utils';
import Event from '../Event';

describe('<Event />', () => {
  it('create component Event', () => {
    const { container } = render(<Event
      date="2021-04-01"
      value={50.5}
      measureType="Lb/s"
    />);

    expect(container).toMatchSnapshot();
  });

  it('check event display', () => {
    render(<Event
      date="2021-04-01"
      value={50.5}
      measureType="Lb/s"
    />);

    expect(screen.queryByText('Thu, April 1st 2021')).not.toBeNull();
    expect(screen.queryByText('50.5 Lb/s')).not.toBeNull();
  });
});
