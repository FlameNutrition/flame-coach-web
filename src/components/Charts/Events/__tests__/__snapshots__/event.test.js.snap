// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Event /> create component Event 1`] = `
<div>
  <div
    class="MuiPaper-root complex-swipeable-list-item MuiPaper-outlined"
  >
    <div
      class="complex-swipeable-list-item-label"
    >
      <span
        class="complex-swipeable-list-item-name"
      >
        50.5
         
        Lb/s
      </span>
    </div>
    <div
      class="complex-swipeable-list-item-description"
    >
      Thu, April 1st 2021
    </div>
  </div>
</div>
`;
